pallets/handles/src/lib.rs

```diff
@@ -0,0 +1,671 @@
+			let canonical_handle_str = HandleConverter::convert_to_canonical(&base_handle_str);
+			let canonical_handle = canonical_handle_str.as_bytes().to_vec();
+			Some(HandleResponse { base_handle, suffix, canonical_handle })
+		}
+
+		/// Get the next available suffixes for a given handle.
+		///
+		/// This function takes a `Vec<u8>` handle and generates the next available
+		/// suffixes for that handle. The number of suffixes to generate is determined
+		/// by the `count` parameter, which is of type `u16`.
+		///
+		/// # Arguments
+		///
+		/// * `handle` - The handle to generate the next available suffixes for.
+		/// * `count` - The number of suffixes to generate.
+		///
+		/// # Returns
+		///
+		/// * `PresumptiveSuffixesResponse` - The response containing the next available suffixes.
+		/// ```
+		pub fn get_next_suffixes(handle: Vec<u8>, count: u16) -> PresumptiveSuffixesResponse {
+			let mut suffixes: Vec<HandleSuffix> = vec![];
+			let base_handle: Handle = handle.try_into().unwrap_or_default();
+			let base_handle_str = core::str::from_utf8(&base_handle).unwrap_or("");
+
+			let canonical_handle_str = HandleConverter::convert_to_canonical(&base_handle_str);
+			let canonical_handle_vec = canonical_handle_str.as_bytes().to_vec();
+			let canonical_handle: Handle = canonical_handle_vec.try_into().unwrap();
+			let suffix_index =
+				Self::get_next_suffix_index_for_canonical_handle(canonical_handle.clone())
+					.unwrap_or_default();
+
+			// Generate suffixes from the next available suffix index
+			let mut suffix_generator = SuffixGenerator::new(
+				T::HandleSuffixMin::get() as usize,
+				T::HandleSuffixMax::get() as usize,
+				&canonical_handle_str,
+			);
+			let sequence = suffix_generator.suffix_iter().enumerate();
+
+			for (i, suffix) in sequence {
+				if i >= suffix_index as usize && i < (suffix_index as usize + count as usize) {
+					suffixes.push(suffix as u16);
+				}
+				if suffixes.len() == count as usize {
+					break
+				}
+			}
+			let response =
+				PresumptiveSuffixesResponse { base_handle: base_handle.into(), suffixes };
+			response
+		}
+
+		/// Retrieve a `MessageSourceId` for a given handle.
+		///
+		/// # Arguments
+		///
+		/// * `display_handle` - The full display handle to retrieve the `MessageSourceId` for.
+		///
+		/// # Returns
+		///
+		/// * `Option<MessageSourceId>` - The `MessageSourceId` if the handle is valid.
+		///
+		pub fn get_msa_id_for_handle(display_handle: Handle) -> Option<MessageSourceId> {
+			let full_handle_str = core::str::from_utf8(&display_handle).unwrap_or("");
+			let (base_handle_str, suffix) = HandleConverter::split_display_name(full_handle_str);
+			let canonical_handle_str = HandleConverter::convert_to_canonical(&base_handle_str);
+			let canonical_handle = canonical_handle_str.as_bytes().to_vec();
+			let canonical_handle: Handle = canonical_handle.try_into().unwrap();
+			let msa = Self::get_msa_id_for_canonical_and_suffix(canonical_handle, suffix);
```

Comment:
You can just return the result of `Self::get_msa_id_for_canonical_and_suffix(canonical_handle, suffix)`

Reply:
Fixed in #1392